src/compiler/checker.ts

```diff
@@ -14085,13 +14085,15 @@ namespace ts {
                     const returnType = getReturnTypeOfSignature(s);
                     return !(returnType.flags & (TypeFlags.Any | TypeFlags.Never)) && checkTypeRelatedTo(returnType, target, relation, /*errorNode*/ undefined);
                 })) {
-                    const resultObj: { errors?: Diagnostic[] } = errorOutputContainer || {};
-                    checkTypeAssignableTo(source, target, node, headMessage, containingMessageChain, resultObj);
-                    const diagnostic = resultObj.errors![resultObj.errors!.length - 1];
+                    const [sourceType, targetType] = getTypeNamesForErrorDisplay(source, target);
+                    const diagnostic = error(node, Diagnostics.Type_0_is_not_assignable_to_type_1, sourceType, targetType);
                     addRelatedInfo(diagnostic, createDiagnosticForNode(
                         node,
                         signatures === constructSignatures ? Diagnostics.Did_you_mean_to_use_new_with_this_expression : Diagnostics.Did_you_mean_to_call_this_expression
                     ));
+                    if (errorOutputContainer && errorOutputContainer.skipLogging) {

```

Comment:
I'm not really familiar with `errorOutputContainer`, but it sort of looks like this should have an else case that adds the diagnostic to `diagnostics` instead.  @sandersn should be able to confirm.

Reply:
In the else clause, you'd add the diagnostic to the headMessage or containingMessageChain, which is was `checkTypeAssignableTo` does. The logic there is somewhat complicated so I'm not sure it's a good idea to replicate it here either. You still need to call `checkTypeAssignableTo`.